app/src/lib/helpers/linux.ts

```diff
@@ -0,0 +1,85 @@
+import { join } from 'path'
+import { pathExists as pathExistsInternal } from 'fs-extra'
+import {
+  ChildProcess,
+  spawn,
+  SpawnOptionsWithoutStdio,
+  SpawnOptions,
+} from 'child_process'
+
+export function isFlatpakBuild() {
+  return __LINUX__ && process.env.FLATPAK_HOST === '1'
+}
+
+/**
+ * Convert an executable path to be relative to the flatpak host
+ *
+ * @param path a path to an executable relative to the root of the filesystem
+ *
+ */
+function convertToFlatpakPath(path: string) {
+  return join('/var/run/host', path)
+}
+
+/**
+ * Checks the file path on disk exists before attempting to launch a specific shell
+ *
+ * @param path
+ *
+ * @returns `true` if the path can be resolved, or `false` otherwise
+ */
+export async function pathExists(path: string): Promise<boolean> {
+  if (isFlatpakBuild()) {
+    path = convertToFlatpakPath(path)
```

Comment:
yup this is correct. the path outside the container should be real but detection for binaries in `/usr` should be pointed to `/var/run/host`

Reply:
I think `/var/lib/flatpak` might be blocked by flatpak not sure(I'll check and update this), `/snap`  is one that won't work since this dir won't be mounted

Reply:
yeah `/var/lib/flatpak` is blocked

Reply:
👍🏻 I'm assuming that the `try-catch` below will ensure these errors will be handled and swallowed by the app, so I'm going to try and avoid getting too complex with the logic in here for now for paths that we know are inaccessible in the flatpak context.